cf_remote/spawn.py

```diff
@@ -308,47 +377,73 @@ def spawn_vm_in_aws(
     size=None,
     role=None,
 ):
-    if platform not in aws_platforms:
-        raise ValueError("Platform '%s' does not exist. (Available platforms: %s)" % (platform,
-                         ", ".join(cloud_data.aws_platforms.keys())))
+    platform_name = platform.split("-")[0]
+    if platform_name not in aws_image_criteria:
+        raise ValueError(
+            "Platform '%s' is not in our set of image criteria. (Available platforms: %s)"
+            % (platform, ", ".join(cloud_data.aws_image_criteria.keys()))
+        )
     try:
         driver = get_cloud_driver(Providers.AWS, aws_creds, region)
         existing_vms = driver.list_nodes()
     except InvalidCredsError as error:
         raise ValueError(
-            "Invalid credentials, check cloud_config.json (%s.)" % str(error)[1:-1])
+            "Invalid credentials, check cloud_config.json (%s.)" % str(error)[1:-1]
+        )
     if name is None:
         name = _get_unused_name(
             [vm.name for vm in existing_vms], platform, _NAME_RANDOM_PART_LENGTH
         )
     else:
         if any(vm.state in (0, "running") and vm.name == name for vm in existing_vms):
             raise ValueError("VM with the name '%s' already exists" % name)
-    aws_platform = aws_platforms[platform]
-    size = size or aws_platform.get("xlsize") or aws_platform["size"]
-    user = aws_platform.get("user")
-    ami = aws_platform["ami"]
-
-    log.info("Spawning new '%s' VM in AWS (AMI: %s, size=%s)" % (platform, ami, size))
-    node = driver.create_node(
-        name=name,
-        image=NodeImage(id=ami, name=None, driver=driver),
-        size=NodeSize(
-            id=size,
-            name=None,
-            ram=None,
-            disk=None,
-            bandwidth=None,
-            price=None,
-            driver=driver,
-        ),
-        ex_keyname=key_pair,
-        ex_security_groups=security_groups,
-        ex_metadata={
-            "created-by": "cf-remote",
-            "owner": whoami(),
-        },
+    criteria = _get_image_criteria(platform)
+    architecture = criteria["architecture"] or aws_defaults["architecture"]
+    sizes = "sizes" in criteria and criteria["sizes"] or aws_defaults["sizes"]
+    small = sizes[architecture]["size"]
+    large = sizes[architecture]["xlsize"]
+    if size == None:
+        if role == "hub":
+            size = large or small
+        else:
+            size = small or large
```

Comment:
I feel like this reads better.
```suggestion
        size = (large or small) if (role == "hub") else (small or large)
```
But it's only personal preference ;)